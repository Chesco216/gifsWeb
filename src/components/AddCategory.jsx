import '../styles.css'
import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  
  const [inputVal, setInputVal] = useState('')

  const onInputChanged = ({ target }) => {
    setInputVal(target.value)
    console.log(inputVal)
  }
  
  const onSubmit = ( event ) => {
    event.preventDefault()
    
    if ( inputVal.trim().length <= 1 ) return;
    
    onNewCategory(inputVal.trim())
    setInputVal('')
  }

  return (
    <form onSubmit={ onSubmit }>
      <input
        type='text'
        placeholder='pipipi'
        value={ inputVal }
        onChange={ onInputChanged }
      />
    </form>
  )
}
