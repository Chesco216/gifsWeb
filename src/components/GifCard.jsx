import React from 'react'
import "../styles.css";

export const GifCard = ({ title, url }) => {
  return (
    <div className='card'>
      <img src={url.url} alt={title}/>
      <p>{title}</p>
    </div>
  )
}

