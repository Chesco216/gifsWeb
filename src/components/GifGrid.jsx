import PropTypes  from "prop-types";
import { useFetchGifs } from '../hooks/useFetchGifs.js';
// import '../styles.css'
import { GifCard } from "./GifCard.jsx";

export const GifGrid = ({ category }) => {
  
  const { images, isLoading } = useFetchGifs( category )

  return (
    <>
      <h1>{category}</h1>
      {
        // isLoading
        // ? (<h1>Loading...</h1>)
        // : null
        isLoading && (<h1>Loading...</h1>)
      }
      <div className="card-grid">
        {
          images.map(( img ) => (
            <GifCard 
              key={img.key}
              { ...img }
            />
          ))
          // DESESTRUCTURANDO COMO TODO BUEN SEGUIDOR DE FH
          // images.map(({ id, title }) => (
          //   <li key={ id }>{ title }</li>
          // ))

        }
      </div>
    </>
  )
}

GifGrid.propTypes = {

  category: PropTypes.string.isRequired

}
