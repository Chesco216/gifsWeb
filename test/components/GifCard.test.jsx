import { render, screen } from "@testing-library/react"
import { GifCard } from "../../src/components/GifCard";

describe( 'GifCard test', () => {

  const title = 'pipipi'
  const url = 'yipee'

  test('should match snapshot', () => {
    
    const { container } = render(<GifCard url={url} title={title}/>)
    expect( container ).toMatchSnapshot()

  })

  test('should show url and alt', () => {

    render(<GifCard title={title} url={url}/>)
    // console.log(screen.debug())
    // expect( screen.getByRole('img').src ).toBe(url)
    expect( screen.getByRole('img').alt ).toBe(title)

  })

})
