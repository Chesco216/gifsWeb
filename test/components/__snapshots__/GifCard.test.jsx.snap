// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifCard test should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="pipipi"
      src="yipee"
    />
    <p>
      pipipi
    </p>
  </div>
</div>
`;
